Practice 1 MPU6050

D-Alexandr opened the pull request:
Added first part practice 1 mpu6050. Script communicates to mpu6050 by i2c and display value temperature on Celsius and Fahrenheit

D-Alexandr committed:
Practice 1 Part 1: Added detect mpu6050 script

A script has been written to search for connected mpu6050 modules.
Enabled debug mode

Signed-off-by: Alexandr Datsenko <datsenkoalexander@gmail.com>

D-Alexandr committed:
Practice 1 Part 1: Added read WHO_AM_I registr

Added variable registers. Reading data from the register WHO_AM_I

Signed-off-by: Alexandr Datsenko <datsenkoalexander@gmail.com>

D-Alexandr committed:
Practice 1 Part 1: Added module enable functionality

Added a variable enable register and send the module enable command

Signed-off-by: Alexandr Datsenko <datsenkoalexander@gmail.com>

D-Alexandr committed:
Practice 1 Part 1: Added temperature reading command

Added variables for high and low temperature registers. Written temperature reading cycle

Signed-off-by: Alexandr Datsenko <datsenkoalexander@gmail.com>

D-Alexandr committed:
Practice 1 Part 1: Added display temperaturs in celsius

Added conversion unsigned to signed value. Added conversion value
ADC to celsius. Added delay 1 second

Signed-off-by: Alexandr Datsenko <datsenkoalexander@gmail.com>

D-Alexandr committed:
Practice 1 Part 1: Added display temperature in fahrenheit

Added conversion celsius to fahrenheit. Disabled debug mode

Signed-off-by: Alexandr Datsenko <datsenkoalexander@gmail.com>

yekovalyov commented on 04-mpu6050/scripts/mpu6050_temp.sh:
```diff
@@ -26,14 +24,17 @@ while [ true ]
     temp_sig=$(((temp_adc_unsig & 0x7fff)- (temp_adc_unsig & 0x8000)))
     echo "Value ADC: $temp_sig"
 
     temp_c=$(echo " scale = 2; $temp_sig / 340 + 36.53" | bc)
```
Required resolution is 0.001. scale=2 provides only resolution 0.01

yekovalyov commented on 04-mpu6050/scripts/mpu6050_temp.sh:
```diff
@@ -26,14 +24,17 @@ while [ true ]
     temp_c=$(echo " scale = 2; $temp_sig / 340 + 36.53" | bc)
     echo "Value temperature in Celsius: $temp_c"
 
+    temp_f=$(echo " scale = 2; (9/5 * $temp_c) + 32" | bc)
```
Required resolution is 0.001. scale=2 provides only resolution 0.01

D-Alexandr commented:
mpu6050 sensitivity to temperature ±1/С°. Why do we need measurements with an accuracy of 0.001?

yekovalyov commented:
> mpu6050 sensitivity to temperature ±1/С°. Why do we need measurements with an accuracy of 0.001?

Please read task and driver docs.

D-Alexandr committed:
Practice 1 Part 1: Fixed temperature calculation accuracy

Changed the accuracy parameter of the calculation. The scale has been changed from 2 to 3

Signed-off-by: Alexandr Datsenko <datsenkoalexander@gmail.com>

D-Alexandr committed:
Practice 1 Part 2: Added Makefile and skeleton for driver

Writed makefile for mpu6050 and skeleton for driver

Signed-off-by: Alexandr Datsenko <datsenkoalexander@gmail.com>

yekovalyov reviewed (changes requested):
Issues in script implementation were addressed.
Only skeleton of module was implemented.

D-Alexandr committed:
Practice 1 Part 2: Added example driver mpu6050

Downloaded exemple driver mpu6050 from github

Signed-off-by: Alexandr Datsenko <datsenkoalexander@gmail.com>

D-Alexandr committed:
Practice 1 Part 2: Fixed errors in driver mpu6050 from repository

Created struct class_attribute for sys interface in driver mpu6050

Signed-off-by: Alexandr Datsenko <datsenkoalexander@gmail.com>

D-Alexandr committed:
Practice 1 Part 2: Added measure temperature in F

Added measure temperature in F. with precision 0.001 F

Signed-off-by: Alexandr Datsenko <datsenkoalexander@gmail.com>

yekovalyov commented on 04-mpu6050/module/mpu6050.c:
```diff
@@ -181,7 +182,7 @@ static ssize_t temp_show(struct class *class,
 	mpu6050_read_data();
 
-	sprintf(buf, "%d\n", g_mpu6050_data.temperature);
+	sprintf(buf, "%d.%03d\n", g_mpu6050_data.temperature / shift_number, g_mpu6050_data.temperature % shift_number);
```
Please check your code style using script from Linux repository.
scripts/checkpatch.pl --no-tree -f *.c

D-Alexandr committed:
Practice 1 Part 2: Fixed code style

Removed space in line number 53

Signed-off-by: Alexandr Datsenko <datsenkoalexander@gmail.com>
